File: Observer pattern/observer C#/WeatherData.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace ObserverPattern
{
    class WeatherData : ISubject
    {
        private List<IObserver> ObserverList;
        private float temperature,humidity,pressure;

        public WeatherData()
        {
            ObserverList = new List<IObserver>();
        }
    
        public void registerObserver(IObserver observer)
        {
            ObserverList.Add(observer);
        }

        public void removeObserver(IObserver observer)
        {
            try
            {
                ObserverList.Remove(observer);
            }
            catch (Exception ex)
            {

            }
        }

        public void notifyObserver()
        {
            for (int i = 0; i < ObserverList.Count; i++)
                ObserverList[i].update(temperature, humidity, pressure);
        }

        public void setMeasurements(float temperature, float humidity, float pressure)
        {
            this.temperature = temperature;
            this.humidity = humidity;
            this.pressure = pressure;
            this.measurementChanged();
        }


        public void measurementChanged()
        {
            this.notifyObserver();
        }
    }
}

File: Observer pattern/observer py/WeatherData.py
############ author charlie ##############
############ concrete class of Subject ##############

from Subject import ISubject
from Observe import IObserver

class WeatherData(ISubject):

	def __init__():
		self.ObserverList=[]
		self.temprature=0.0
		self.humidity=0.0
		self.pressure=0.0
	
	def registerObserver(self,observer):###### observer type : IObserver
		ObserverList.append(observer)
	def removeObserver(self,observer):  ###### observer type : IObserver
		observer=ObserverList.index(observer)
		if not observer:
			ObserverList.remove(observer)
	def notifyObserver(self):
		for ob in ObserverList:
			ob.update(self.temprature,self.humidity,self.pressure)
	def getTemperature(self):
		pass
	def getHumidity(self):
		pass
	def getPressure(self):
		pass
	def measurementChanged(self):
		self.notifyObserver()
	def setMeasurements(self,temperature,humudity,pressure):
		self.temperature=temperature
		self.humudity=humudity
		self.pressure=pressure
		self.measurementChanged()
